apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/FeatureFlagsButton.tsx

```diff
@@ -179,63 +184,68 @@ const FeatureFlagsButton = () => {
-                          disabled for current language
-                        </Tag>
-                      ) : null}
-                    </Flex>
-                  </Button>
-                  {focusedName === flagName ? (
-                    <Flex flexDirection="column" pl={6} rowGap={3}>
-                      <EditSection
-                        value={inputValues[flagName] || JSON.stringify(featureFlags[flagName])}
-                        disabled={!inputValues[flagName]}
-                        error={error}
-                        onChange={handleInputChange}
-                        onOverride={handleOverrideFeature}
-                        onRestore={handleRestoreFeature}
-                      />
-                      <Flex p={3} backgroundColor="neutral.c30">
-                        <Text whiteSpace="pre">
-                          {JSON.stringify(featureFlags[flagName], null, 2)}
-                        </Text>
+              {filteredFlags.map(([flagName, value]) => {
+                const { overriddenByEnv, overridesRemote, ...pureValue } = value; // eslint-disable-line
+                return (
+                  <>
+                    <OldButton
+                      flexDirection="row"
+                      py={1}
+                      onClick={() => {
+                        flagName === focusedName ? setFocusedName() : setFocusedName(flagName);
+                      }}
+                    >
+                      <Flex flex={1} mr={3} alignItems="center">
+                        <Box
+                          bg={value?.enabled ? "success.c100" : "error.c100"}
+                          height={10}
+                          width={10}
+                          mr={2}
+                          borderRadius={999}
+                        />
+                        <Text mr={1}>{flagName}</Text>
+                        {value?.overriddenByEnv ? (
+                          <Tag active mx={1} type="opacity" size="small">
+                            overridden by env
+                          </Tag>
+                        ) : value?.overridesRemote ? (
+                          <Tag active mx={1} type="opacity" size="small">
+                            overridden locally
+                          </Tag>
+                        ) : null}
+                        {value?.enabledOverriddenForCurrentLanguage ? (
+                          <Tag active mx={1} type="outlinedOpacity" size="small">
+                            disabled for current language
+                          </Tag>
+                        ) : null}
+                      </Flex>
+                    </OldButton>
+                    {focusedName === flagName ? (
+                      <Flex flexDirection="column" pl={6} rowGap={3}>
+                        <EditSection
+                          value={inputValues[flagName] || JSON.stringify(pureValue)}
```

Comment:
prefer to cache this with a memo. `stringify` every time it re-renders isn't ideal. you may want to have an intermediary component to ease this as it's included in a "if", generally speaking the JSX tree is quite deep here and splitting into more components isn't bad.

Reply:
Thanks I'll do that 👍

Reply:
Done 👌
I split it in 3 components on both LLM & LLD -> `FeatureFlagsSettings/index.tsx` `FeatureFlagDetails.tsx` `FeatureFlagEdit.tsx`.
It's much clearer now I think